refactor: cleanup interviews

krantheman opened the pull request:
### Issue

Closes: https://github.com/frappe/hrms/issues/781

### Changes

- Redid Feedback section of the Interview DocType
  - Eliminated data duplication by removing average rating, result, and other fields from the `Interview Detail` table. This table only has an Interviewer link field now.
  - Added a Feedback tab that summarizes and displays data directly from `Interview Feedback`
- Made `Interview Type` optional for `Interview Round`
- Updated applicable interviewers to default to `Interview Round` Interviewers
- Updated Submit Feedback button in Interview to be disabled, but visible, for unauthorized users (as opposed to not being rendered at all)
- Fixed issue where cancellation of an individual `Interview Feedback` required all docs of the same Interview to be cancelled
- Removed unnecessary overlap validation to allow for more flexibility
- Updated tests to reflect changes
- Updated Appraisal -> Feedback tab to use the same feedback summary & history templates as Interview -> Feedback

![image](https://github.com/frappe/hrms/assets/61287991/96db6cb2-437f-4ea1-8e5e-131c835c728c)

### Documentation

https://frappehr.com/docs/v14/en/interview-management

krantheman committed:
refactor(Interview Round): make Interview Type optional

krantheman committed:
feat: add Feedback Tab

krantheman committed:
feat: pull feedback data from server

krantheman committed:
feat: add ratings summary

krantheman committed:
refactor: calculate reviews per rating based on user average rating

krantheman committed:
feat: add individual ratings

krantheman committed:
refactor: clean up feedback summary

krantheman committed:
feat: add circular progress bar

krantheman committed:
feat: dynamic circular progress bar

krantheman committed:
refactor: get appllicable interviewers

krantheman committed:
feat: use circular progress bar for skills

krantheman committed:
feat: consider colors for ratings between 0 and 1

krantheman committed:
fix: circular progress size

krantheman committed:
refactor: disable Submit Feedback for unauthorized users

krantheman committed:
feat: update interview average rating on submitting feedback

krantheman committed:
refactor: clean up interview_details

krantheman committed:
test: clean up instances of interview_detail

krantheman committed:
chore: delete redundant doctype Interview Detail

krantheman committed:
refactor: better querying

krantheman committed:
fix: Feedback rendering of new Interviews

krantheman committed:
fix: No feedback message

krantheman committed:
fix: rounding

krantheman committed:
chore: remove unused import

ruchamahabal commented on hrms/public/scss/interview.scss:
```diff
@@ -0,0 +1,89 @@
+.circular-progress{
+    width: 100px;
```
use tabs. Check all your HTML files also

krantheman replied:
I am SO sorry

krantheman committed:
fix: Feedback Summary subtext

krantheman committed:
chore: convert spaces to tabs ...again

krantheman committed:
chore: improve readability

ruchamahabal commented:
- make the feedback cards tighter - reduce padding
- star distribution chart - remove colors - convert to gray (like appraisal feedback summary)
- Font size for chart labels & rating - looks bigger than feedback font
- Rating summary header font sizes should be consistent with section heading sizes
- Tighter spacing in summary - between "based on x reviews" and the stars

**Extra:**
- Reuse/change appraisal feedback template
- Reduce spacing between section title & description

<img width="1057" alt="image" src="https://github.com/frappe/hrms/assets/24353136/bc525e48-0db1-4d74-9815-d888a3705945">

krantheman committed:
style: change rating colors to gray

krantheman committed:
fix: font size and spacing

krantheman committed:
refactor: performance feedback summary

krantheman committed:
refactor: use common feedback_history and feedback_summary

krantheman committed:
fix(Appraisal): Employee Performance Feedback link

krantheman committed:
refactor: clean up feedback_history context

krantheman committed:
chore: rename performance_feedback_history to performance_feedback

krantheman committed:
fix(circular_progress): file name typo

krantheman committed:
feat: add Interviewers field which automatically fetches interviewers based on Interview Round

krantheman committed:
refactor: consider Interviewers selected in Interview rather than from Interview Round

krantheman committed:
refactor: use Table instead of Table MultiSelect for Interviewers

krantheman committed:
chore: merge with develop and fix conflicts

krantheman committed:
chore: convert map to list comprehension

krantheman committed:
test: fix Interviewers

krantheman committed:
chore: fix formatting

ruchamahabal committed:
Merge branch 'develop' into cleanup-interviews

ruchamahabal committed:
Merge branch 'develop' into cleanup-interviews

ruchamahabal committed:
refactor: retain interview detail doctype

- to avoid breaking change

ruchamahabal committed:
fix: fetching applicable interviewers for interview & feedback

ruchamahabal committed:
refactor: move fetch fields to `fetch_from` config, cleanup js controller

ruchamahabal committed:
fix: move whitelisted getter functions outside class

- `run_doc_method` on cancelled document errors out

ruchamahabal committed:
perf: remove lazy-loaded bundles from app include js

- add common html templates in main bundle

ruchamahabal committed:
fix: run feedback functions serially

ruchamahabal committed:
fix: don't break whitelisted API

- retain interviewer mapping in child table in APIs

ruchamahabal committed:
test: fix interview tests

ruchamahabal committed:
feat: show hover message for disabled feedback submission

ruchamahabal committed:
chore: cleanup templates

ruchamahabal committed:
feat: set current interview timings as default while rescheduling

- user has to enter all fields again

- before sending rescheduling email check if timings have changed

ruchamahabal committed:
fix: notify update after updating average rating in interview

ruchamahabal committed:
fix: move Reschedule Interview to Actions

ruchamahabal committed:
fix: use left join in interview feedback query to fetch feedback from non-employee users

ruchamahabal committed:
test(APIs): `get_feedback` & `get_skill_wise_average_rating`

ruchamahabal committed:
chore: remove unused col break

ruchamahabal commented on hrms/hr/doctype/interview/interview.json:
```diff
@@ -174,6 +168,26 @@
+   "fieldname": "interview_details",
+   "fieldtype": "Table",
+   "label": "Interviewers",
+   "options": "Interview Detail"
```
```suggestion
   "options": "Interview Detail",
   "allow_on_submit": 1
```

ruchamahabal committed:
chore: allow updating interview detail on submission

ruchamahabal commented:
@Mergifyio backport version-15-hotfix

ruchamahabal commented:
@Mergifyio copy version-15-hotfix

ruchamahabal commented:
@krantheman can you open a PR on v15 hotfix
